### PR TITLE
Add initial challenge template

### PR DESCRIPTION
**Pull request by celinval**
Add an initial template for challenges, and also a CI status badge for README and tools page.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: doc/src/template.md [hunk 1]
```diff
@@ -1 +1,66 @@
-# Challenge Template
+# Challenge XXXX[^challenge_id]: Challenge Title
+
+- **Status:** *One of the following: [Open | Resolved | Expired]*
+  - **Solution:** *Option field to point to the PR that solved this challenge.*
```
Comment (tautschnig) [inline doc/src/template.md:4]: Is this indentation intentional?

File: doc/src/template.md [hunk 2]
```diff
@@ -1 +1,66 @@
-# Challenge Template
+# Challenge XXXX[^challenge_id]: Challenge Title
```
Comment (tautschnig) [inline doc/src/template.md:1]: Is "Challenge" the right wording?

File: doc/src/template.md [hunk 3]
```diff
@@ -1 +1,66 @@
...
+- **End date:** *YY/MM/DD*
+
+-------------------
+
+
+Status: [Open | Resolved | Expired]:
+Tracking issue:
+Start date:
+Due date:
+Author(s):
```
Comment (tautschnig) [inline doc/src/template.md:16]: This somehow overlaps with the lines above?!

File: doc/src/template.md [hunk 4]
```diff
@@ -1 +1,66 @@
...
+
+*Explain why this is a challenge that should be prioritized. Consider using a motivating example.*
+
+## Description
+
+*Describe the challenge in more details.*
+
+### Assumptions
+
+*Mention any assumption that users may make. Example, "assuming the usage of Stack Borrow".*
```
Comment (tautschnig) [inline doc/src/template.md:33]: s/Stack Borrow/Stacked Borrows/

File: doc/src/template.md [hunk 5]
```diff
@@ -1 +1,66 @@
...
+|	|	|
+|	|	|
+|	|	|
+
+All proofs must automatically ensure the absence of the following undefined behaviors [[ref]](https://github.com/rust-lang/reference/blob/142b2ed77d33f37a9973772bd95e6144ed9dce43/src/behavior-considered-undefined.md):
+
+*List of UBs*
+
+Note: All solutions to verification challenges need to satisfy the criteria established in the [challenge book](general-rules.md)
+in addition to the ones listed below
```
Comment (tautschnig) [inline doc/src/template.md:64]: What is listed below?!

File: doc/src/template.md (hunk 2 above)
Comment (celinval) [reply doc/src/template.md:1]: Any suggestions? We could use `Verification Task` or something like that.

File: doc/src/template.md (hunk 1 above)
Comment (celinval) [reply doc/src/template.md:4]:
```suggestion
- **Solution:** *Option field to point to the PR that solved this challenge.*
```

File: doc/src/template.md (hunk 3 above)
Comment (celinval) [reply doc/src/template.md:16]: My bad. :)

Comment (celinval) [inline doc/src/template.md:16]:
```suggestion
```

File: doc/src/template.md (hunk 4 above)
Comment (celinval) [reply doc/src/template.md:33]: Why do I keep making the same mistake? :disappointed:

Comment (celinval) [inline doc/src/template.md:33]:
```suggestion
*Mention any assumption that users may make. Example, "assuming the usage of Stacked Borrows".*
```

File: doc/src/template.md (hunk 5 above)
Comment (celinval) [reply doc/src/template.md:64]:
```suggestion
in addition to the ones listed above
```
:smile:

File: doc/src/template.md [hunk 6]
```diff
@@ -1 +1,59 @@
...
+
+*Describe the challenge in more details.*
+
+### Assumptions
+
+*Mention any assumption that users may make. Example, "assuming the usage of Stacked Borrows".*
+
+### Success Criteria*
+
+*Here are few possible criteria:*
```
Comment (tautschnig) [inline doc/src/template.md:30]: How about "Here are some examples of possible criteria:" (else at least insert "a" before "few possible")

File: doc/src/template.md [hunk 7]
```diff
@@ -1 +1,59 @@
...
+|	|	|
+|	|	|
+|	|	|
+
+All proofs must automatically ensure the absence of the following undefined behaviors [[ref]](https://github.com/rust-lang/reference/blob/142b2ed77d33f37a9973772bd95e6144ed9dce43/src/behavior-considered-undefined.md):
+
+*List of UBs*
+
+Note: All solutions to verification challenges need to satisfy the criteria established in the [challenge book](general-rules.md)
+in addition to the ones listed above
```
Comment (tautschnig) [inline doc/src/template.md:57]: Add a full stop at the end of the sentence.

**Review by tautschnig — APPROVED** [review]

File: doc/src/template.md (hunk 6 above)
Comment (celinval) [inline doc/src/template.md:30]:
```suggestion
*Here are some examples of possible criteria:*
```

File: doc/src/template.md (hunk 7 above)
Comment (celinval) [reply doc/src/template.md:57]:
```suggestion
in addition to the ones listed above.
```

**Pull request merged**